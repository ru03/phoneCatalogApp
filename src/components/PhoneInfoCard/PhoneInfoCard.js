import React from 'react';
import classes from './PhoneInfoCard.module.css';

const phoneInfoCard = (props) => (
    <article className={classes.Card}>
        <header className={classes.Header}>
            <img src="https://vignette.wikia.nocookie.net/battlefordreamisland/images/c/cd/IPhone_X.png/revision/latest?cb=20180625201840"></img>
            <h1>{props.name}</h1>
        </header>
        <section className={classes.Section}>
            <p>Model: {props.desc}</p>
            <button>Detalles</button>
        </section>
    </article>
)

export default phoneInfoCard;