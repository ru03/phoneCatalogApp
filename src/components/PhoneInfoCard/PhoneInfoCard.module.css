.Card {
    box-shadow: 0 2px 1px -1px rgba(0,0,0,.2), 
                0 1px 1px 0 rgba(0,0,0,.14),
                0 1px 3px 0 rgba(0,0,0,.12);
    max-width: 400px;
    background-color: white;
    color: black;
    margin: 10px;
}

.Header {
    display: flex;
    flex-flow: column;
    align-items: center;
}

.Section {
    display: flex;
    flex-direction: column;
    margin: 10px;
}

img {
    max-width: 90%;
}