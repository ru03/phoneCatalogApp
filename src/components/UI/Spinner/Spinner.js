import React from 'react';
import classes from './Spinner.module.css';
import Backdrop from './../Backdrop/Backdrop';
const spinner = (props) => (
    <>
        <Backdrop></Backdrop>
        <div className={classes.spinner}>
            <div className={classes.doubleBounce1}></div>
            <div className={classes.doubleBounce2}></div>
        </div>
    </>
)

export default spinner;