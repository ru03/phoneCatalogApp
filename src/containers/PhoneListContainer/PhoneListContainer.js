import React, { Component } from 'react';
import { connect } from 'react-redux';
import * as actions from '../../store/actions/index';

import PhoneInfoCard from '../../components/PhoneInfoCard/PhoneInfoCard';
import classes from './PhoneListContainer.module.css';

class PhoneListContainer extends Component {

    componentDidMount () {
        this.props.onFetchPhones();
    }

    render() {
        let data = this.props.phones.map(details => <PhoneInfoCard  key={details.id} 
                                                                    name={details.Brand}
                                                                    image={details.img}
                                                                    model={details.DeviceName}></PhoneInfoCard>)
        return (
            <div className={classes.Items}>
                {data}
            </div>
        );

    }

}

const mapStateToProps = state => {
    console.log(state)
    return {
        phones: state.phones,
        loading: state.loading
    };
};

const mapDispatchToProps = dispatch => {
    return {
        onFetchPhones: () => dispatch(actions.fetchPhones())
    };
};


export default connect(mapStateToProps, mapDispatchToProps)(PhoneListContainer);