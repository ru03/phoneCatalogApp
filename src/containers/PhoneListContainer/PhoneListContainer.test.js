import React from 'react';
import { mount } from 'enzyme';
import { PhoneListContainer } from './PhoneListContainer';
import PhoneInfoCard from './../../components/PhoneInfoCard/PhoneInfoCard';
import PhoneDetails from './../../components/PhoneDetailComponent/PhoneDetailComponent';
import Spinner from './../../components/UI/Spinner/Spinner';

describe('PhoneListContainer test', () => {
    let phoneListContainer;
    beforeEach(() => {
        const props = {
            phones: [],
            phone: [],
            loading: false
        }
        phoneListContainer = mount(<PhoneListContainer onFetchPhones={() => {}} {...props}/>);
    });

    it('PhoneListContainer component should be displayed', () => {
        expect(phoneListContainer).toMatchSnapshot();
    });

    it('PhoneListContainer component should display no items message without phones', () => {
        expect(phoneListContainer.find('.Items').text()).toEqual('No items were found');
    });

    it('PhoneListContainer component should display spinner if loading is true', () => {
        phoneListContainer.setProps({
            loading: true
        });
        expect(phoneListContainer.find(Spinner)).toHaveLength(1);
    });

    it('PhoneListContainer component should display PhoneInfoCard component', () => {
        phoneListContainer.setProps({
            phones: [{
                id: 1,
                model: 'iphone8',
                price: '15',
                colors: 'red',
                brand: 'apple',
                screen: 'screen',
                image: 'image'
            }]
        });
        expect(phoneListContainer.find(PhoneInfoCard)).toHaveLength(1);
    });

    it('PhoneListContainer component should display PhoneDetails with a phone in store', () => {
        phoneListContainer.setProps({
            phones: [{
                id: 1,
                model: 'iphone8',
                price: '15',
                colors: 'red',
                brand: 'apple',
                screen: 'screen',
                image: 'image'
            }],
            phone: [{
                id: 1,
                model: 'iphone8',
                price: '15',
                colors: 'red',
                brand: 'apple',
                screen: 'screen',
                image: 'image'
            }]
        });
        expect(phoneListContainer.find(PhoneDetails)).toHaveLength(1);
    });
});