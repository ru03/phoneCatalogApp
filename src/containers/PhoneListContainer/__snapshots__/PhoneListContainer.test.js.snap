// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhoneListContainer test PhoneListContainer component should be displayed 1`] = `
<PhoneListContainer
  loading={false}
  onFetchPhones={[Function]}
  phone={Array []}
  phones={Array []}
>
  <div
    className="Items"
  >
    No items were found
  </div>
  <div />
</PhoneListContainer>
`;
